source/adios2/engine/dataman/DataManWriter.cpp

```diff
@@ -150,15 +150,10 @@ void DataManWriter::MetadataThread(const std::string &address)
     while (m_Listening)
     {
         auto request = tpm.ReceiveRequest();
-        if (request == nullptr)
+        if (request && request->size() > 0)
         {
-            continue;
-        }
-        if (request->size() >= 0)
-        {
-            m_AggregatedMetadataMutex.lock();
+            std::lock_guard lck(m_AggregatedMetadataMutex);
```

Comment:
Looks like the template argument doesn't resolve by default so you need `std::lock_guard<std::mutex> lck`